packages/gateway/src/metrics.js

```diff
@@ -91,23 +102,81 @@ export async function metricsGet(request, env, ctx) {
     `# TYPE nftgateway_requests_total counter`,
     ...env.ipfsGateways.map(
       (gw) =>
-        `nftgateway_requests_total{gateway="${gw}",env="${env.ENV}"} ${
-          metricsCollected.ipfsGateways[gw].totalSuccessfulRequests +
-          metricsCollected.ipfsGateways[gw].totalFailedRequests
-        }`
+        `nftgateway_requests_total{gateway="${gw}",env="${env.ENV}"} ${totalResponsesPerGateway[gw]}`
     ),
     `# HELP nftgateway_successful_requests_total Total successful requests performed to each gateway.`,
     `# TYPE nftgateway_successful_requests_total counter`,
     ...env.ipfsGateways.map(
       (gw) =>
-        `nftgateway_successful_requests_total{gateway="${gw}",env="${env.ENV}"} ${metricsCollected.ipfsGateways[gw].totalSuccessfulRequests}`
+        `nftgateway_successful_requests_total{gateway="${gw}",env="${
+          env.ENV
+        }"} ${
+          metricsCollected.ipfsGateways[gw].totalResponsesByStatus[
+            HTTP_SUCCESS_CODE
+          ] || 0
```

Comment:
@alanshaw decided to track this individually to have Prometheus metrics to be persisted with a different key to success to have a prometheus name tag, and only use the other key with "failed requests"

If you disagree, happy to change it to simply `nftgateway_requests_by_status_total` and have everything under the same name. 

The main reason for this decision was that I wanted to have a metric that includes a count of all failed requests here, and for queries to Prometheus seemed more reasonable to have success separate. Lmk what you think

Reply:
I think it'll be unexpected to not see 200 in `nftgateway_requests_by_status_total`...

If you want to have a success vs failure metric for convenience then that's fine but I'd still include 200 in `nftgateway_requests_by_status_total`. You can match a tag via regexp in prometheus so it should be easy to group failure like `/^[45][0-9][0-9]$/` and success like `/^2[0-9][0-9]$/` or something. I won't make it _loads_ easier. Personally I wouldn't complicate it here but it's up to you.

The only other concern with explicitly pulling out 200 is I don't know that _all_ successfull requests from the gateway are 200...but I guess probably yes.

Reply:
Yes, but my current naming is `nftgateway_failed_requests_by_status_total`. Anyway, I will just remove failed and add 200 then.